include/envoy/common/crypto/crypto.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Common {
+namespace Crypto {
+
+class CryptoObject {
+public:
+  virtual ~CryptoObject() = default;
+};
+
+using CryptoObjectPtr = std::unique_ptr<CryptoObject>;
+
+namespace Access {
+
+template <class T> T* getTyped(CryptoObjectPtr* cryptoPtr) {
```

Comment:
don't take raw pointer of a smart pointer anywhere, use raw pointer of the object, or smart pointer itself.

for here take `CryptoObject&` as parameter.

Reply:
@bdecoste `CryptoObject&`, not `CryptoObjectPtr&` please...